### PR TITLE
MULE-11214: Add concept of "resolverName" to the NamedTypeResolver

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/metadata/BaseWscResolver.java [hunk 1]
```diff
@@ -37,6 +38,11 @@ public String getCategoryName() {
     return WSC_CATEGORY;
   }
 
+  @Override
+  public String getResolverName() {
+    return WSC_NAME;
```
Comment (juandesi) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/metadata/BaseWscResolver.java]: move this to the InputResolver and change it's value for "ConsumeInputResolver"

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/BooleanKeyResolver.java [hunk 2]
```diff
@@ -52,6 +49,7 @@ public String getCategoryName() {
     return categoryName;
   }
 
+
```
Comment (juandesi) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/BooleanKeyResolver.java]: :trollface:

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidator.java [hunk 3]
```diff
@@ -56,11 +60,15 @@
  */
 public class MetadataComponentModelValidator implements ModelValidator {
 
+  public static final String EMPTY_RESOLVER_NAME = "%s '%s' specifies a metadata resolver [%s] which has an empty %s name";
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidator.java]: private

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidator.java [hunk 4]
```diff
@@ -79,10 +87,41 @@ private void validateComponent(ComponentModel model) {
...
   }
 
+  private void validateResolversName(ComponentModel model, MetadataResolverFactory resolverFactory,
+                                     Table<String, String, Class<?>> names) {
+    List<NamedTypeResolver> resolvers = new LinkedList<>();
+    resolvers.addAll(getAllInputResolvers(model, resolverFactory));
+    resolvers.add(resolverFactory.getOutputResolver());
+
+    resolvers.stream()
+        .filter(r -> !NULL_RESOLVER_NAME.equals(r.getResolverName()))
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/MetadataComponentModelValidator.java]: this shouldn't depend on the name. Shouldn't you have a NullResolver class?

**Pull request merged**